omero_marshal/__init__.py

```diff
@@ -43,6 +45,22 @@ def get_decoder(t):
         return None
 
 
+def get_schema_version(version):
+    p = re.compile('^(\d+\.\d+\.\d+).*')
```

Comment:
Trailing `.*` not required.

Also, I find that having regexes available as package variables so that they can be tested explicitly very helpful. You are already testing the bounds and errors conditions of the regex via your `get_schema_version()` test cases but it can be useful to do regex testing in isolation. Especially when you inevitably have to go back months or years later and remember what you were building the regex to match.